src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckJavaAgent.kt

```diff
@@ -106,18 +107,33 @@ internal object LincheckJavaAgent {
         // allowing already loaded classes re-transformation.
         instrumentation.addTransformer(LincheckClassFileTransformer, true)
         // The transformation logic depends on the testing strategy.
-        // In the stress testing mode, Lincheck needs to track coroutine suspensions,
-        // so it processes all classes (including those that are already loaded),
-        // and looks for suspension points. In case of the model checking, Lincheck
-        // could also process all the classes, but it would lead to a significant
-        // performance degradation. Instead, in the model checking mode, Lincheck
-        // processes classes lazily, only when they are used. However, we have an
-        // option to enable the global transformation in the model checking mode
-        // for testing purposes.
-        if (instrumentationMode == STRESS || INSTRUMENT_ALL_CLASSES_IN_MODEL_CHECKING_MODE) {
-            // Re-transform the already loaded classes.
-            // New classes will be transformed automatically.
-            instrumentation.retransformClasses(*getLoadedClassesToInstrument().toTypedArray())
+        when {
+            // If an option to enable transformation of all classes is explicitly set,
+            // then we re-transform all the classes
+            // (this option is used for testing purposes).
+            INSTRUMENT_ALL_CLASSES -> {
+                // Re-transform the already loaded classes.
+                // New classes will be transformed automatically.
+                instrumentation.retransformClasses(*getLoadedClassesToInstrument().toTypedArray())
+            }
+
+            // In the stress testing mode, Lincheck needs to track coroutine suspensions.
+            // As an optimization, we remember the set of loaded classes that actually
+            // have suspension points, so later we can re-transform only those classes.
+            instrumentationMode == STRESS -> {
+                check(instrumentedClasses.isEmpty())
+                val classes = getLoadedClassesToInstrument().filter {
+                    // new classes that were loaded after the latest STRESS mode re-transformation
+                    !transformedClassesStress.containsKey(it.name) ||
+                    // old classes that were already loaded before and have coroutine method calls inside
```

Comment:
It might be the first time when a stress test is running, so `coroutineCallingClasses` is still empty. Therefore, you can ignore some classes that should not be ignored.

Reply:
Yes, in this case `transformedClassesStress` would be empty, and the check `!transformedClassesStress.containsKey(it.name)` will return true.